### PR TITLE
squid:S3052 - Fields should not be initialized to default values

### PR DESCRIPTION
**Pull request by m-ezzat**
This pull request is focused on resolving occurrences of Sonar rule squid:S3052 - Fields should not be initialized to default values

You can find more information about the issue here: https://dev.eclipse.org/sonar/coding_rules#q=squid:S3052

Please let me know if you have any questions.

M-Ezzat

### CODE REVIEW

File: src/main/java/org/pircbotx/Configuration.java [hunk 1]
```diff
@@ -245,23 +245,23 @@ protected Configuration(Builder builder) {
 		/**
 		 * Enable or disable sending WEBIRC line on connect, default disabled
```
Comment (SSederberg) [inline src/main/java/org/pircbotx/Configuration.java]: If you're going to remove the defaults and just keep it blank, the code documentation must reflect that.

Comment (m-ezzat) [inline src/main/java/org/pircbotx/Configuration.java]: What would you suggest to be written ?

Comment (lilggamegenius) [inline src/main/java/org/pircbotx/Configuration.java]: just remove the ", default disabled" text?

Comment (m-ezzat) [inline src/main/java/org/pircbotx/Configuration.java]: Ok, I was just making sure.

Comment (SSederberg) [inline src/main/java/org/pircbotx/Configuration.java]: Yes, that's what I mean't, and you need to do that with the rest of Configuration.java

Comment (m-ezzat) [inline src/main/java/org/pircbotx/Configuration.java]: ok, I've pushed the new changes, just let me know if there's anything else.

**Comment by m-ezzat** [comment]
I am not sure whether you do not agree with these changes or not. Please close this PR if not.

**Comment by hoerup** [comment]
How would you determine how a Configuration element behaves if there's not set a default AND the javadoc doesn't describe default behaviour ?

**Pull request closed without merging**